ansible/configs/ocp-clientvm/env_vars.yml

```diff
@@ -177,3 +180,31 @@ instances:
       - device_name: "{{docker_device}}"
         volume_size: 100
         volume_type: gp2
+    security_groups:
+      - BastionSG
+
+#### OSP ####
+# See cloud_providers/osp_default_vars.yml
+# See roles/infra-osp-project-create/defaults/main.yml
+
+# Set this to true if you need to create a new project in OpenStack
+# This should almost always be set to true for OpenShift installations
+# If it is set to false, the {{ osp_project_name }} must already exist and
+# should be able to run whatever you are deploying
+osp_project_create: true
```

Comment:
I think this is still something to be discussed. 

Why don't we create a common project for client VMs. There is no reason that I can think of to have a separate project for each. It is just going to be a single VM with a single disk deployed on a single (existing) network with a single FIP.

Reply:
I don't know if that's how we will do.
I like the separation per project. TBD with our OSP team.

But if we choose to use one project for a kind of deployment, when deploying from CloudForms + agnosticV we can hardcode the project name:

```yaml
osp_project_name: LABS-clientvms
admin_user: opentlc-mgr
# set quotas globally for this catalog item  
quota_num_instances: 200
...
```

We don't even have to set `osp_project_create: false`, since it should be idem potent.

From an agnosticD user/dev point of view, it makes things easier since everything is done by ansible. I like to minimize the amount of manual things to do.

If by default we set `osp_project_create` to false, then we have to tell in the documentation to login in OpenStack, create a project, etc...

Reply:
Actually, since deletion is impacted by this choice, i made some changes:
- always delete heat stack  (first step)
- delete resources of the project only if `osp_project_create` is `true`.

@stencell let me know if you can think of any problem when deleting the heat stack first.

and let me update my previous comment:

> But if we choose to use one project for a kind of deployment, when deploying from CloudForms + agnosticV we can hardcode the project name:
```yaml
osp_create_project: false
osp_project_name: LABS-clientvms
admin_user: opentlc-mgr
# set quotas globally for this catalog item  
quota_num_instances: 200
```
> We **have to set** `osp_project_create: false`, because when deleting, we don't want to delete everything in the project.

Reply:
> Actually, since deletion is impacted by this choice, i made some changes:
> 
> * always delete heat stack  (first step)
> * delete resources of the project only if `osp_project_create` is `true`.
> 
> @stencell let me know if you can think of any problem when deleting the heat stack first.
> 

I think this will cause a problem. The heat stack creates resources like network and router. Over the course of the lifetime of a project, users may create instances, ports, etc. that are outside of the heat stack. If you try to tear the heat stack down without deleting those extra ports, for instance, the heat stack teardown will fail.

This is why i originally did the purge(s) first. Then deleting the heat stack cleaned up the remainder that was originally created. Deleting the project always comes at the end regardless of the order of the previous two.

@fridim

Reply:
OK will change